Add tutorial to restore rds db snapshot to dbinstance/cluster (#1396)
aws-controllers-k8s/community: docs/content/docs/tutorials/rds-example.md

EOF
```

> ## Restore Snapshot to DBInstance or DBCluster

Reviewer: I would suggest "Restore a Database Snapshot" as the title. We can cover the semantics of `DBInstace` v `DBCluster` in the language below.

You can restore a snapshot to DBInstance or DBCluster with setting `SnapshotIdentifier` inside `DBCluster` or `DBSnapshotIdentifier` inside `DBInstance` CRD to restore the snapshot to a DBCluster or DBInstance. 

> `SnapshotIdentifier` need match the existing DBCluster snapshot identifier or ARN of DBInstance snapshot, `DBSnapshotIdentifier` need match the identifier of an existing DBSnapshot. 

Reviewer 2: ```suggestion
`SnapshotIdentifier` should match an existing DBCluster snapshot identifier or  the ARN of a DBInstance snapshot, `DBSnapshotIdentifier` should match the identifier of an existing DBSnapshot. 
```

> You can restore a snapshot to DBInstance or DBCluster with setting `SnapshotIdentifier` inside `DBCluster` or `DBSnapshotIdentifier` inside `DBInstance` CRD to restore the snapshot to a DBCluster or DBInstance. 
> `SnapshotIdentifier` need match the existing DBCluster snapshot identifier or ARN of DBInstance snapshot, `DBSnapshotIdentifier` need match the identifier of an existing DBSnapshot. 
> Once it's set and resource created, update them will have no effect. You can use the following example to restore a DBSnapshot to DBCluster: 

Reviewer: ```suggestion
You can also restore a database snapshot to a specific `DBInstance` or `DBCluster` using the ACK for RDS controller.

To restore a database snapshot to a `DBInstance`, you must set the `DBSnapshotIdentifier` parameter. `DBSnapshotIdentifier` need match the identifier of an existing DBSnapshot.

To restore a database snapshot to a `DBCluster`, you must set the `SnapshotIdentifier`. The value of `SnapshotIdentifier` must match either an existing `DBCluster` snapshot identifier or an ARN of a `DBInstance`snapshot.

The following examples show how you can restore database snapshots both to `DBCluster` and `DBInstance` resources: 
```

Reviewer: I think we also need a line on where you can find the snapshot identifier.

> Once it's set and resource created, update them will have no effect. You can use the following example to restore a DBSnapshot to DBCluster: 

Reviewer 2: ```suggestion
Once it's set and the resource is created, updating the `SnapshotIdentifer` or `DBSnapshotIdentifier` fields will have no effect.

Use the following example to to create a DBCluster from a DBSnapshot:
```

```bash
cat <<EOF > rds-restore-dbcluster-snapshot.yaml
apiVersion: rds.services.k8s.aws/v1alpha1
kind: DBCluster
metadata:
  name: "${RDS_CLUSTER_NAME}"
spec:
  dbClusterIdentifier: "${RDS_CLUSTER_NAME}"
  engine: aurora-postgresql
  engineVersion: "13.7"

>   snapshotIdentifier: arn:aws:rds:${RDS_REGION}:${RDS_CUSTOMER_ACCOUNT}:snapshot:${RDS_DB_SNAPSHOT_IDENTIFIER}

Reviewer: Where is `RDS_DB_SNAPSHOT_IDENTIFIER` defined?

EOF

kubectl apply -f rds-restore-dbcluster-snapshot.yaml
```

> Example to restore a DBSnapshot to DBInstance:

Reviewer 2: ```suggestion
Use the following example to create a DBInstance from a DBSnapshot:
```

```bash
cat <<EOF > rds-restore-dbinstance-snapshot.yaml
apiVersion: rds.services.k8s.aws/v1alpha1
kind: DBInstance
metadata:
  name: "${RDS_INSTANCE_NAME}"
spec:
  allocatedStorage: 20
  dbInstanceClass: db.m5.large
  dbInstanceIdentifier: "${RDS_INSTANCE_NAME}"
  engine: postgres
  engineVersion: "14"
  masterUsername: "postgres"
  multiAZ: true

>   dbSnapshotIdentifier: "${RDS_DB_SNAPSHOT_IDENTIFIER}"

Reviewer: Where is Where is `RDS_DB_SNAPSHOT_IDENTIFIER` defined?

EOF
```

## Restore a Database Snapshot

You can also restore a database snapshot to a specific `DBInstance` or `DBCluster` using the ACK for RDS controller.

> To restore a database snapshot to a `DBInstance`, you must set the `Spec.DBSnapshotIdentifier` parameter. `Spec.DBSnapshotIdentifier` should match the identifier of an existing DBSnapshot.

Reviewer: Looking at this again, if we use the `spec` prefix we should follow the convention from earlier in the tutorial:

`spec.dbSnapshotIdentifier`.

Author: I will follow the convention to remove the `Spec`here
